staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -220,7 +223,13 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 	}
 
 	if rules.WarnIfAllMissing && len(missingList) > 0 && len(kubeconfigs) == 0 {
-		klog.Warningf("Config not found: %s", strings.Join(missingList, ", "))
+		missingString := strings.Join(missingList, ", ")
+
+		if rules.WarnIfAllMissingCallback != nil {
```

Comment:
Here if no `WarbubgHandler` is defined we'd silently ignore the problem. The original issue was to entirely get rid of warnings from library code.

Reply:
is kubectl able to set the warning handler to preserve visibility to the failure when running with kubectl? Shouldn't we be setting that?

Reply:
yeah, it should be possible to do, that's why I'm advocating having this be settable.

Reply:
Should I use `klog.Infof` by default instead of `klog.Warningf`?